modules/apps/bookmarks/bookmarks-web/src/main/resources/META-INF/friendly-url-routes/routes.xml

```diff
@@ -15,6 +15,18 @@
 		<implicit-parameter name="p_p_state">maximized</implicit-parameter>
 	</route>
 	<route>
+		<pattern>/{fullInstanceId}/folder/{folderId:\d+}</pattern>
```

Comment:
How come you didn't use {fullInstancePath}?
Using {fullInstanceId} will not be enough for portlets for which it might be an empty String such as non-instanceable portlets since you would end up with a double bar.